src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -445,7 +446,58 @@ protected static void ShiftFieldsFromRootToInstallerLevel(InstallerManifest inst
                     property.SetValue(installerManifest, null);
                 }
             }
-        }
+        }
+
+        /// <summary>
+        /// Shifts common installer fields from installer level to manifest root.
+        /// </summary>
+        /// <param name="installerManifest">Wrapper object containing the installer manifest object models.</param>
+        protected static void ShiftInstallerFieldsToRootLevel(InstallerManifest installerManifest)
+        {
+            var rootProperties = installerManifest.GetType().GetProperties();
+            var installerProperties = installerManifest.Installers.First().GetType().GetProperties();
+
+            // Get common properties between root and installer level
+            var commonProperties = rootProperties.Where(rp => installerProperties.Any(ip => ip.Name == rp.Name));
+
+            foreach (var property in commonProperties)
+            {
+                var firstInstallerValue = installerManifest.Installers.First().GetType().GetProperty(property.Name).GetValue(installerManifest.Installers.First());
+                if (firstInstallerValue != null)
+                {
+                    // Check if all installers have the same value
+                    bool allInstallersHaveSameValue = installerManifest.Installers.All(i =>
+                    {
+                        var propertyValue = i.GetType().GetProperty(property.Name).GetValue(i);
+                        return propertyValue != null &&
+                        propertyValue.Equals(firstInstallerValue);
+                    });
+
+                    // If value is false, it can be becuase we don't have .Equals() override implemented for the type of the property.
+                    // For that, we check further if the property is of type list and check if the lists are equal
+                    if (!allInstallersHaveSameValue)
+                    {
+                        if (firstInstallerValue is IList installerValueList)
+                        {
+                            allInstallersHaveSameValue = installerManifest.Installers.All(i =>
+                            i.GetType().GetProperty(property.Name).GetValue(i) is IList otherList &&

```

Comment:
Do we need to check if `i.GetType().GetProperty(property.Name).GetValue(i)` is null before checking if it is of IList type?

Reply:
if (null is IList installerValueList) will return false and the flow will move to the else statement that'll return false. So, there's no need to explicitly check for null here